core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -296,6 +332,8 @@ private[spark] class Executor(
       } else {
         taskRunner.kill(interruptThread = interruptThread, reason = reason)
       }
+      // Safe to remove kill mark as we got a chance with the TaskRunner.
+      killMarks.remove(taskId)
```

Comment:
Add the remove in a try/finally

Reply:
You need try/finally for a `ConcurrentHashMap.remove` operation??

Reply:
Start `try` at killMarks.put, and the remove goes in the finally

Reply:
Ok, ignore this - intellij was messing up the braces ... sigh